apl-core/src/main/java/com/apollocurrency/aplwallet/apl/core/chainid/BlockchainConfigUpdater.java

```diff
@@ -105,4 +122,67 @@ private HeightConfig getConfigAtHeight(int targetHeight) {
             .map(height -> new HeightConfig(blockchainProperties.get(height)))
             .orElse(null);
     }
+
+    public HeightConfig getCurrentConfig() {
+        return blockchainConfig.getCurrentConfig();
+    }
+
+    /**
+     * Return correct ShardingSetting by specified (trim) height. We find any shard settings with height
+     * associated with height.
+     *
+     * @param trimHeight target height (trim height usually)
+     * @return found ShardingSettings value OR Optional.Empty value if not found
+     */
+    public Optional<ShardingSettings> getShardingSettingsByTrimHeight(int trimHeight) {
+        if (this.chain == null) {
+            String error = "Can't get 'ShardingSettings', because Chain configuration is not initialized ! That's strange actually...";
+            log.warn(error);
+            return Optional.empty();
+        }
+        Map<Integer, BlockchainProperties> blockchainProperties = chain.getBlockchainProperties();
+        if (blockchainProperties == null) { // very small chance, but who knows...?
+            String error = String.format("Missing any 'BlockchainProperties' by trimHeight '%s' !", trimHeight);
+            log.warn(error);
+            return Optional.empty();
+        }
+        if (trimHeight < 0) { // no chance to look for negative height value in configs
```

Comment:
Not needed throwing exception, because caller expects empty value.

Reply:
Here you write error message to the log, but caller code is not aware of such error, so that you allow caller code to successfully proceed and hide possible error until some more significant error occurs.